metric_learn/base_metric.py

```diff
@@ -28,7 +18,7 @@ def transformer(self):
     -------
     L : upper triangular (d x d) matrix
     """
-    return cholesky(self.metric()).T
+    return cholesky(self.metric_).T
```

Comment:
You're right, it should not be there. I was waiting to implement ``ExplicitMixin`` so that I could move them there, and I left it here to pass tests but I should rather have copied it in every Metric Learner class to be coherent with inheritance. Anyway now that we've decided to implement everything in ``MahalanobisMixin`` for now (I updated the PR message),  I'll just need to move ``transformer`` as well as ``transform`` in ``MahalanobisMixin``.